table.go

```diff
@@ -1113,6 +1113,12 @@ func (t *Table) externalParquetCompaction(writer io.Writer) func(compact []*part
 // compactParts will compact the given parts into a Parquet file written to w.
 // It returns the size in bytes of the compacted parts.
 func (t *Table) compactParts(w io.Writer, compact []*parts.Part) (int64, error) {
+	block, done, err := t.ActiveWriteBlock()
```

Comment:
It seems like the issue here is that we close a table before persisting to object storage, which means that this `ActiveWriteBlock` will fail with `ErrTableClosing` when persisting on close. How did we get around this before?

Reply:
I think the answer is that we had some compaction-like code that did not acquire the write lock on the block

Reply:
Didn't it acquire the read lock instead? I'd have to look again, but I'm pretty sure that's the piece I overlooked

Reply:
No, it was a write block (I'm pretty sure it has to right?) 
https://github.com/polarsignals/frostdb/blob/c9100f2ac9c7aca111e1362be4a8a67b85b6b44b/compaction.go#L154

The compaction on serialization was different code: https://github.com/polarsignals/frostdb/blob/cbeaf0d100edc9476bb27b013cbd023f02c78ced/table.go#L1172

Reply:
Updated this PR to fall back to the previous behavior on `ErrTableClosing` but I'm curious: should we be acquiring a write block earlier on (e.g. in merge) since we'll be updating the index?